src/savegame_legacy.cpp

```diff
@@ -2054,16 +2054,19 @@ std::istream& operator>>(std::istream& is, SkillLevel& obj) {
 
 void monster::load_legacy(std::stringstream & dump) {
     int idtmp, plansize, speed, faction_dummy;
-    dump >> idtmp >> position.x >> position.y >> wandx >> wandy >> wandf >> moves >> speed >>
-         hp >> sp_timeout[0] >> plansize >> friendly >> faction_dummy >> mission_id >>
-         no_extra_death_drops >> dead >> anger >> morale;
+    dump >> idtmp >> position.x >> position.y >> wander_pos.x >> wander_pos.y >> wandf
+         >> moves >> speed >> hp >> sp_timeout[0] >> plansize >> friendly
+         >> faction_dummy >> mission_id >> no_extra_death_drops >> dead >> anger >> morale;
 
+    wander_pos.z = g->get_levz();
+    position.z = g->get_levz();
```

Comment:
Yes, I did. That was a blind quick fix (I'd have to recompile the whole game to test it and I'm still cleaning up AIM).